webapp/src/projects.tsx

```diff
@@ -422,26 +423,22 @@ export class ProjectsCarousel extends data.Component<ProjectsCarouselProps, Proj
                             />
                         )}
                     </carousel.Carousel>
-                    <div ref="detailView" className={`detailview ${cards.filter((scr, index) => index == selectedIndex).length > 0 ? 'visible' : ''}`}>
-                        {cards.filter((scr, index) => index == selectedIndex).length > 0 ?
-                            <div className="close">
-                                <sui.Icon tabIndex={0} aria-hidden={false} icon="remove circle" onClick={this.closeDetail} />
-                            </div> : undefined}
-                        {cards.filter((scr, index) => index == selectedIndex).map(scr =>
-                            <ProjectsDetail parent={this.props.parent}
-                                name={scr.name}
-                                key={'detail' + scr.name}
-                                description={scr.description}
-                                url={scr.url}
-                                imageUrl={scr.imageUrl}
-                                largeImageUrl={scr.largeImageUrl}
-                                youTubeId={scr.youTubeId}
-                                scr={scr}
+                    <div ref="detailView" className={`detailview ${selectedElement ? 'visible' : ''}`}>
+                        {selectedElement && <sui.CloseButton onClick={this.closeDetail}/>}
```

Comment:
small: could pull this 'selectedElement' out and wrap the detailView instead of each child element